eng/targets/Imports.targets

```diff
@@ -228,6 +228,29 @@
     <None Include="$(MSBuildThisFileDirectory)\..\..\src\Setup\PowerShell\uninstall.ps1" PackagePath="tools\uninstall.ps1" Visible="false" Pack="true"/>
   </ItemGroup>
 
+  <!-- RestrictedInternalsVisibleTo -->
+  <ItemDefinitionGroup>
+    <RestrictedInternalsVisibleTo>
+      <Visible>false</Visible>
+    </RestrictedInternalsVisibleTo>
+  </ItemDefinitionGroup>
+
+  <ItemGroup>
+    <InternalsVisibleTo Include="@(RestrictedInternalsVisibleTo)" Key="%(Key)" />
+  </ItemGroup>
+
+  <Target Name="PrepareGenerateRestrictedInternalsVisibleTo"
+          BeforeTargets="PrepareGenerateInternalsVisibleToFile"
+          Condition="'@(RestrictedInternalsVisibleTo)' != ''">
+    <ItemGroup>
+      <_InternalsVisibleToAttribute Include="System.Runtime.CompilerServices.RestrictedInternalsVisibleToAttribute">
+        <_Parameter1>%(RestrictedInternalsVisibleTo.Identity)</_Parameter1>
+        <_Parameter2 Condition="'%(RestrictedInternalsVisibleTo.Partner)' != ''">Microsoft.CodeAnalysis.ExternalAccess.%(RestrictedInternalsVisibleTo.Partner).Api</_Parameter2>
+        <_Parameter2 Condition="'%(RestrictedInternalsVisibleTo.Partner)' == ''">Microsoft.CodeAnalysis.ExternalAccess.UnknownPartner.Api</_Parameter2>
```

Comment:
><_Parameter2 Condition="'%(RestrictedInternalsVisibleTo.Partner)' == ''">Microsoft.CodeAnalysis.ExternalAccess.UnknownPartner.Api</_Parameter2> [](start = 8, length = 143)

Should this rather report an error?

Reply:
I'm not sure this will be used enough to matter.

Reply:
I'll add a warning here.